### PR TITLE
a8n: Delete expired CampaignPlans in background

### PR DESCRIPTION
**Pull request by mrnugget**
A `CampaignPlan` is expired when

- all its `CampaignJobs` have finished execution
- they have finished > 1 hour ago
- the `CampaignPlan` has not been attached to a `Campaign`

In the background of the enterprise `repo-updater` (since its a single instance) now runs a goroutine that deletes these expired `CampaignPlan`s every 2 minutes.

What's still missing: the tests that make sure a `CampaignPlan` is not deleted when attached to a `Campaign`. That depends on changes to the `a8n.Store` that @tsenart made in another branch that hasn't been merged yet. But... the test is also not really strictly necessary, so, ready for review.

### CODE REVIEW

File: enterprise/pkg/a8n/store.go [hunk 1]
```diff
@@ -1232,6 +1232,49 @@ var deleteCampaignPlanQueryFmtstr = `
...
+  campaigns.campaign_plan_id = campaign_plans.id
+)
+AND
+NOT EXISTS (
+  SELECT id
+  FROM
+  campaign_jobs
+  WHERE
+  campaign_jobs.campaign_plan_id = campaign_plans.id
+  AND
```
Comment (slimsag) [inline enterprise/pkg/a8n/store.go]: I am an SQL n00b so just taking a guess: If you changed this to OR, could ditch the `SELECT 1` on lines 1255-1261?

Comment (slimsag) [reply enterprise/pkg/a8n/store.go]: Oh nevermind, I missed that this selects on `campaign_jobs` but the other selects on `campaigns`

File: enterprise/cmd/repo-updater/main.go [hunk 2]
```diff
@@ -19,13 +21,33 @@ func main() {
...
+
+		for {
+			log15.Info("DeleteExpiredCampaignPlans running")
+
+			err := store.DeleteExpiredCampaignPlans(ctx)
+			if err != nil {
+				log15.Error("DeleteExpiredCampaignPlans", "error", err)
+			}
+
+			<-ticker.C
```
Comment (tsenart) [inline enterprise/cmd/repo-updater/main.go]: Do we want to make this wait 2 minutes only after the query ran? (i.e. with `time.Sleep`)

File: enterprise/pkg/a8n/store_test.go [hunk 3]
```diff
@@ -1444,5 +1444,104 @@ func testStore(db *sql.DB) func(*testing.T) {
 				}
 			}
 		})
+
+		t.Run("CampaignPlan DeleteExpired", func(t *testing.T) {
```
Comment (tsenart) [inline enterprise/pkg/a8n/store_test.go]:
```suggestion
		t.Run("CampaignPlan/DeleteExpired", func(t *testing.T) {
```

File: enterprise/cmd/repo-updater/main.go (hunk 2 above)
Comment (mrnugget) [reply enterprise/cmd/repo-updater/main.go]: Ah, you're right. I had a `for { select { } }` here before...

**Pull request merged**